src/storage/engine/mod.rs

```diff
@@ -231,6 +233,8 @@ mod tests {
                 .unwrap();
             must_put_cf(e.as_ref(), "cf", b"k", b"v1");
         }
+        // Sleep a while to make sure the `LOCK` file is deleted.
+        thread::sleep(Duration::from_millis(100));
```

Comment:
This error occurs in your PR, not in master. 
I guess, In your PR, even the threadpool dropped, the DB may can't be dropped until the thread terminates. 

/cc @BusyJay https://github.com/frewsxcv/rust-threadpool/blob/master/lib.rs#L361
If the channel has un-handled message but the sender is closed, receiver will return error?